arb_os/inbox.mini

```diff
@@ -157,24 +157,57 @@ public impure func inbox_get() -> IncomingRequest {
         let ebMsg = unsafecast<IncomingRequest>(rawMsg);
         if (ebMsg.blockNumber > globalInbox.blockNum) {
             if ((globalInbox.sequencer != None<SequencerState>) && globalInbox.seenMsgAtThisBlockNum) {
-                let eobMsg = endOfBlockMessage(globalInbox.blockNum, globalInbox.timestamp);
-                globalInbox = globalInbox with {
-                    blockNum: ebMsg.blockNumber
-                } with {
-                    timestamp: ebMsg.timestamp
-                } with {
-                    seenMsgAtThisBlockNum: false
-                };
-                return eobMsg;
+                // don't update queue, because we're not consuming ebMsg
+                let rawSideloadMsg = asm(globalInbox.blockNum,) any { sideload };
+                if (rawSideloadMsg == ()) {
+                    let eobMsg = endOfBlockMessage(globalInbox.blockNum, globalInbox.timestamp);
+                    globalInbox = globalInbox with {
+                        blockNum: ebMsg.blockNumber
+                    } with {
+                        timestamp: ebMsg.timestamp
+                    } with {
+                        seenMsgAtThisBlockNum: false
+                    };
+                    return eobMsg;
+                } else {
+                    let newMsgRaw = unsafecast<IncomingRequestFromInbox>(rawSideloadMsg);
+                    let newMsg = struct {
+                        kind: newMsgRaw.kind,
+                        blockNumber: newMsgRaw.blockNumber,
+                        timestamp: newMsgRaw.timestamp,
+                        sender: newMsgRaw.sender,
+                        requestId: newMsgRaw.requestId,
+                        msgData: bytearray_fromSizeAndBuffer(newMsgRaw.msgSize, newMsgRaw.msgData),
+                        provenance: struct {
+                            l1SeqNum: newMsgRaw.requestId,
+                            parentRequestId: 0,
+                            indexInParent: ~0,
+                        },
+                        aggregator: None<AggregatorInfo>,
+                        adminMode: true,
+                    };
+
+                    if (newMsg.blockNumber == 0) {
+                        newMsg = newMsg with { blockNumber: globalInbox.blockNum };
+                    }
+                    if (newMsg.timestamp == 0) {
+                        newMsg = newMsg with { timestamp: globalInbox.timestamp };
+                    }
```

Comment:
Seems like there's a lot of duplication between here and the sideload block below. Would be good to refactor into a common function probably, possible a function to convert `IncomingRequestFromInbox` to `IncomingRequest` could be used in four different places in this file

Reply:
Factored this out into a common function.